sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -540,6 +540,12 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       }
     }
   }
+
+  test("SPARK-31580: Read a file written before ORC-569") {
```

Comment:
This test case is in `OrcSuite` and will be tested in both `OrcSourceSuite` and `HiveOrcSourceSuite`.